tests/test_stability_funs.py

```diff
@@ -349,3 +351,402 @@ def test_magwell_print(capsys):
+
+        iota = jnp.mean(data0["iota"])
+        shear = jnp.mean(data0["shear"])
+        psi = jnp.mean(data0["psi"])
+        sign_psi = jnp.sign(psi)
+        sign_iota = jnp.sign(iota)
+
+        phi = zeta
+
+        B = grid.meshgrid_reshape(data0["|B|"], "arz")
+        B_sup_zeta = grid.meshgrid_reshape(data0["B^zeta"], "arz")
+        gradpar = B_sup_zeta / B
+
+        dpdpsi = jnp.mean(mu_0 * data0["p_r"] / data0["psi_r"])
+
+        g_sup_aa = grid.meshgrid_reshape(data0["g^aa"], "arz")[None, ...]
+        g_sup_ra = grid.meshgrid_reshape(data0["g^ra"], "arz")[None, ...]
+        g_sup_rr = grid.meshgrid_reshape(data0["g^rr"], "arz")[None, ...]
+
+        gds2 = jnp.reshape(
+            rho**2
+            * (
+                g_sup_aa
+                - 2 * sign_iota * shear / rho * zeta0[:, None, None, None] * g_sup_ra
+                + zeta0[:, None, None, None] ** 2 * (shear / rho) ** 2 * g_sup_rr
+            ),
+            (N_alpha, N_zeta0, N_zeta),
+        )
+
+        f = a_N * B_N**3 * gds2 / B**3 * 1 / gradpar
+        g = a_N**3 * B_N * gds2 / B * gradpar
+        g_half = (g[:, :, 1:] + g[:, :, :-1]) / 2
+
+        cvdrift = grid.meshgrid_reshape(data0["cvdrift"], "arz")[None, ...]
+        cvdrift0 = grid.meshgrid_reshape(data0["cvdrift0"], "arz")[None, ...]
+
+        c = (
+            a_N**3
+            * B_N
+            * jnp.reshape(
+                2
+                / B_sup_zeta[None, ...]
+                * sign_psi
+                * rho**2
+                * dpdpsi
+                * (
+                    cvdrift
+                    - shear / (2 * rho**2) * zeta0[:, None, None, None] * cvdrift0
+                ),
+                (N_alpha, N_zeta0, N_zeta),
+            )
+        )
+
+        h = phi[1] - phi[0]
+
+        A = jnp.zeros((N_alpha, N_zeta0, N_zeta - 2, N_zeta - 2))
+
+        i = jnp.arange(N_alpha)[:, None, None, None]
+        l = jnp.arange(N_zeta0)[None, :, None, None]
+        j = jnp.arange(N_zeta - 2)[None, None, :, None]
+        k = jnp.arange(N_zeta - 2)[None, None, None, :]
+
+        A = A.at[i, l, j, k].set(
+            g_half[i, l, k] / f[i, l, k] / h**2 * (j - k == -1)
+            + (
+                -(g_half[i, l, j + 1] + g_half[i, l, j]) / f[i, l, j + 1] / h**2
+                + c[i, l, j + 1] / f[i, l, j + 1]
+            )
+            * (j - k == 0)
+            + g_half[i, l, j] / f[i, l, j + 1] / h**2 * (j - k == 1)
+        )
+
+        w = jnp.linalg.eigvals(jnp.where(jnp.isfinite(A), A, 0))
+
+        lam1 = jnp.max(jnp.real(jnp.max(w, axis=(2,))))
+
+        # now compute our regular metrics and compare them
+        data_keys = ["ideal ball lambda", "Newcomb ball metric"]
+        data = eq.compute(data_keys, grid=grid)
+
+        lam2 = np.max(data["ideal ball lambda"])
+        Newcomb_metric = data["Newcomb ball metric"]
+
+        np.testing.assert_allclose(lam1, lam2, atol=5e-3, rtol=1e-8)
```

Comment:
Please don't use a finite `atol` and increase `rtol` instead. Otherwise we can't tell if this is a good tolerance or not. If the growth rate is small for this example, the test would always pass even if the calculation is wrong.